causalpy/pymc_models.py

```diff
@@ -117,3 +118,73 @@ def build_model(self, X, y, coords):
             sigma = pm.HalfNormal("sigma", 1)
             mu = pm.Deterministic("mu", pm.math.dot(X, beta), dims="obs_ind")
             pm.Normal("y_hat", mu, sigma, observed=y, dims="obs_ind")
+
+
+class InstrumentalVariableRegression(ModelBuilder):
+    """Custom PyMC model for instrumental linear regression"""
+
+    def build_model(self, X, Z, y, t, coords, priors):
+        """Specify model with treatment regression and focal regression data and priors
+
+        :param X: A pandas dataframe used to predict our outcome y
+        :param Z: A pandas dataframe used to predict our treatment variable t
+        :param y: An array of values representing our focal outcome y
+        :param t: An array of values representing the treatment t of
+                  which we're interested in estimating the causal impact
+        :param coords: A dictionary with the coordinate names for our
+                       instruments and covariates
+        :param priors: An optional dictionary of priors for the mus and
+                      sigmas of both regressions
+
+        e.g priors = {'mus': [[10, 0], [2, 0]], 'sigmas': [[1, 1], [1, 1]]}
+
+        """
+
+        # --- Priors ---
+        with self:
+            self.add_coords(coords)
+            beta_t = pm.Normal(
+                name="beta_t",
+                mu=priors["mus"][0],
+                sigma=priors["sigmas"][0],
+                dims="instruments",
+            )
+            beta_z = pm.Normal(
+                name="beta_z",
+                mu=priors["mus"][1],
+                sigma=priors["sigmas"][1],
+                dims="covariates",
+            )
+            sd_dist = pm.HalfCauchy.dist(beta=2, shape=2)
+            chol, corr, sigmas = pm.LKJCholeskyCov(
+                name="chol_cov", eta=2, n=2, sd_dist=sd_dist
```

Comment:
Do you want to make these prior parameters also customizable through the `coords` mapping?

Reply:
Just want to clear up what you mean here - should I let the coord names and values be changeable or the prior values for the Lkj var. Or both?